ruma-events-macros/src/event.rs

```diff
@@ -371,6 +375,37 @@ fn expand_from_into(
     }
 }
 
+fn expand_eq_ord_event(input: &DeriveInput, fields: &[Field]) -> Option<TokenStream> {
+    if fields.iter().flat_map(|f| f.ident.as_ref()).any(|f| f == "event_id") {
+        let ident = &input.ident;
+        let (impl_gen, ty_gen, where_clause) = input.generics.split_for_impl();
+
+        Some(quote! {
+            impl #impl_gen ::std::cmp::PartialEq for #ident #ty_gen #where_clause {
+                fn eq(&self, other: &Self) -> bool {
```

Comment:
Since we're usually using unambiguous paths, we could use `::std::primitive::bool` here (only matters if the scope the macro is used in defines a different `bool` type).